Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -3132,6 +3146,77 @@ extension DocumentationContext {
         }
         diagnosticEngine.emit(problems)
     }
+        
+    func analyzeAlternateRepresentations() {
+        var problems = [Problem]()
+        for node in topicGraph.nodes.values {
+            guard let entity = try? self.entity(with: node.reference) else { continue }
+            
+            var sourceLanguageToReference: [SourceLanguage: AlternateRepresentation] = [:]
+            for alternateRepresentation in entity.metadata?.alternateRepresentations ?? [] {
+                guard case .resolved(.success(let counterpartReference)) = alternateRepresentation.reference,
+                        let counterpartEntity = try? self.entity(with: counterpartReference) else {
+                    continue
+                }
+
+                // Case where the original symbol already was defined in the languages of the counterpart symbol.
+                let duplicateSourceLanguages = counterpartEntity.availableSourceLanguages.intersection(entity.availableSourceLanguages)
+                if !duplicateSourceLanguages.isEmpty {
+                    problems
+                        .append(
+                            Problem(
```

Comment:
style: I find this style to be a bit too much.
```suggestion
                    problems.append(Problem(
```